test/IECore/PointsAlgoTest.py

```diff
@@ -98,6 +98,22 @@ def testPointsVertexToUniform( self ) :
 		self.assertEqual( p.interpolation, IECore.PrimitiveVariable.Interpolation.Uniform )
 		self.assertEqual( p.data, IECore.FloatVectorData( [ sum(range(0,10))/10. ] ) )
 
+	def testPointsVertexToUniformZeroPointCount( self ) :
+		def makeEmptyPoints() :
+			testObject = IECore.PointsPrimitive( IECore.V3fVectorData() )
+			testObject["b"] = IECore.PrimitiveVariable( IECore.PrimitiveVariable.Interpolation.Vertex, IECore.FloatVectorData() )
+
+			self.assertTrue( testObject.arePrimitiveVariablesValid() )
+			return testObject
+
+		points = makeEmptyPoints()
+
+		p = points["b"]
+		IECore.PointsAlgo.resamplePrimitiveVariable( points, p, IECore.PrimitiveVariable.Interpolation.Uniform )
+
+		self.assertEqual( p.interpolation, IECore.PrimitiveVariable.Interpolation.Uniform )
+		self.assertEqual( p.data, IECore.FloatVectorData( [] ) )
```

Comment:
This isn't anything you've changed in this PR, but I'm surprised that we get FloatVectorData for Uniform data on points. Since there can only be one value, I would expect to get FloatData. Is there a reason we're doing things the way we are?

Reply:
The only argument I think think of is to be consistent with other primitive types where we would expect an array. So our current logic is if we see constant interpolation expect SimpleTypedData otherwise VectorTypedData for all primitive types?

I'll create a PR on cortex 10 and consider the change to return SimpleTypedData there for uniform interpolation of points there if that's ok? Can I proceed to merge this as is on 9_maintenance?

Reply:
Ah, that makes sense. Yeah, merge away. Let's just merge as-is for 10 too - if needs be we can discuss the scalar/array things when Andrew is back, but I think I was probably just overthinking it...